File: implementacao/src/models/entities/Email.ts
class Email {
    constructor(
      email: string
    ) {
        this.email = email;
    }
    email: string;

    validarEmail(){
        //validar email
        const regex : RegExp = /^[a-z0-9.]+@[a-z0-9]+\.[a-z]+\.([a-z]+)?$/i; 
        if(regex.test(this.email)){
            return true
        }else{
            return false
        }
    }
  }
  
  export default Email;

File: app/app/src/main/java/com/app/myapplication/model/entitie/Email.kt
package com.app.myapplication.model.entitie

data class Email(var endereco: String) {
    constructor(): this("luis@email.com")
    private val regex = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+\$"
    fun isValid(): Boolean {
        //return this.endereco.matches(regex.toRegex())
        return true
    }
}